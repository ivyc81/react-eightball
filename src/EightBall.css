.EightBall {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 300px;
    width: 300px;
    border-radius: 50%;
    color: white;
    font-size: 1.5em;
}
