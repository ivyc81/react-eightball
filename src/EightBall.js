import React, { Component } from 'react';
import './EightBall.css';

class EightBall extends Component {
    static defaultState = {
            currentColor: 'black',
            currentMsg: 'Think of a Question.',
    }

    constructor(props){
        super(props);
        this.state = {...EightBall.defaultState}

        this.handleClick = this.handleClick.bind(this);
        this.reset = this.reset.bind(this);
    }

    static defaultProps = {
        answers: [
            { msg: "It is certain.", color: "green" },
            { msg: "It is decidedly so.", color: "green" },
            { msg: "Without a doubt.", color: "green" },
            { msg: "Yes - definitely.", color: "green" },
            { msg: "You may rely on it.", color: "green" },
            { msg: "As I see it, yes.", color: "green" },
            { msg: "Most likely.", color: "green" },
            { msg: "Outlook good.", color: "green" },
            { msg: "Yes.", color: "green" },
            { msg: "Signs point to yes.", color: "goldenrod" },
            { msg: "Reply hazy, try again.", color: "goldenrod" },
            { msg: "Ask again later.", color: "goldenrod" },
            { msg: "Better not tell you now.", color: "goldenrod" },
            { msg: "Cannot predict now.", color: "goldenrod" },
            { msg: "Concentrate and ask again.", color: "goldenrod" },
            { msg: "Don't count on it.", color: "red" },
            { msg: "My reply is no.", color: "red" },
            { msg: "My sources say no.", color: "red" },
            { msg: "Outlook not so good.", color: "red" },
            { msg: "Very doubtful.", color: "red" },
        ]
    }

    /** returns a random answer object from this.props.answers */

    getRandom(){
        return this.props.answers[Math.floor(Math.random()*this.props.answers.length)];
    }

    // getRandomChoice(arr){
    //     return arr[Math.floor(Math.random()*arr)];
    // }

    handleClick(evt){
        const { msg, color} = this.getRandom();

        this.setState({
            currentColor: color,
            currentMsg: msg,
        });
    }

    reset(evt){
        this.setState(EightBall.defaultState);
    }

    render() {
        const style = {backgroundColor: this.state.currentColor};
        return (
            <div>
                <div className="EightBall" onClick={ this.handleClick } style={style}>
                    <b>{ this.state.currentMsg }</b>
                </div>
                <button onClick={this.reset}> Reset </button>
            </div>
        );
    }
}

export default EightBall;