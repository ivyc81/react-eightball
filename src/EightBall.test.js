import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import EightBall from "./EightBall";

// smoke test
it("renders without crashing", function() {
  mount(<EightBall />);
});

// snapshot test
it("matches snapshot", function() {
  let wrapper = mount(<EightBall />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it("renders initial message with correct color", function() {
    let wrapper = mount(<EightBall />);

    let b_elem = wrapper.find('b').first();
    expect(b_elem.equals(<b>Think of a Question.</b>)).toEqual(true);
    expect(wrapper.matchesElement(<div className="EightBall" style="backgroundColor: black" />));
});

it("renders correct message", function() {
    let wrapper = mount(<EightBall a/>);

    wrapper.setState({ currentMsg: "test"});
    let b_elem = wrapper.find('b').first();
    expect(b_elem.equals(<b>test</b>)).toEqual(true);
});

it("changes default message on click", function() {
    const answerArr = [{ msg: "It is certain.", color: "green" }]
    let wrapper = mount(<EightBall answers={answerArr}/>);

    wrapper.simulate("click");
    let b_elem = wrapper.find('b').first();
    expect(b_elem.equals(<b>It is certain.</b>)).toEqual(true);
});

it("changes color on click", function() {
    const answerArr = [{ msg: "It is certain.", color: "green" }]
    let wrapper = mount(<EightBall answers={answerArr}/>);

    wrapper.simulate("click");
    expect(wrapper.matchesElement(<div className="EightBall" style="backgroundColor: green" />));
});

it("changes default message on click", function() {
    const answerArr = [
        { msg: "It is certain.", color: "green" },
        { msg: "It is decidedly so.", color: "green" },
        { msg: "Without a doubt.", color: "green" }
    ]
    let wrapper = mount(<EightBall answers={answerArr}/>);

    wrapper.simulate("click");
    expect(answerArr).toContainEqual({"color": wrapper.state().currentColor, "msg":wrapper.state().currentMsg});
});