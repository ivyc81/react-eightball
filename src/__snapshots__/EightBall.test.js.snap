// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<EightBall
  answers={
    Array [
      Object {
        "color": "green",
        "msg": "It is certain.",
      },
      Object {
        "color": "green",
        "msg": "It is decidedly so.",
      },
      Object {
        "color": "green",
        "msg": "Without a doubt.",
      },
      Object {
        "color": "green",
        "msg": "Yes - definitely.",
      },
      Object {
        "color": "green",
        "msg": "You may rely on it.",
      },
      Object {
        "color": "green",
        "msg": "As I see it, yes.",
      },
      Object {
        "color": "green",
        "msg": "Most likely.",
      },
      Object {
        "color": "green",
        "msg": "Outlook good.",
      },
      Object {
        "color": "green",
        "msg": "Yes.",
      },
      Object {
        "color": "goldenrod",
        "msg": "Signs point to yes.",
      },
      Object {
        "color": "goldenrod",
        "msg": "Reply hazy, try again.",
      },
      Object {
        "color": "goldenrod",
        "msg": "Ask again later.",
      },
      Object {
        "color": "goldenrod",
        "msg": "Better not tell you now.",
      },
      Object {
        "color": "goldenrod",
        "msg": "Cannot predict now.",
      },
      Object {
        "color": "goldenrod",
        "msg": "Concentrate and ask again.",
      },
      Object {
        "color": "red",
        "msg": "Don't count on it.",
      },
      Object {
        "color": "red",
        "msg": "My reply is no.",
      },
      Object {
        "color": "red",
        "msg": "My sources say no.",
      },
      Object {
        "color": "red",
        "msg": "Outlook not so good.",
      },
      Object {
        "color": "red",
        "msg": "Very doubtful.",
      },
    ]
  }
>
  <div
    className="EightBall"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  >
    <b>
      Think of a Question.
    </b>
  </div>
</EightBall>
`;
